sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java

```diff
@@ -92,41 +94,32 @@ final CallGraph constructCompleteCallGraph(View view, List<MethodSignature> entr
    *
    * @return the initialized call graph used in the call graph algorithm
    */
-  protected MutableCallGraph initializeCallGraph() {
-    return new GraphBasedCallGraph();
+  protected MutableCallGraph initializeCallGraph(
+      List<MethodSignature> entryPoints, List<MethodSignature> clinits) {
+    ArrayList<MethodSignature> rootSignatures = new ArrayList<>(entryPoints);
+    rootSignatures.addAll(clinits);
+    return new GraphBasedCallGraph(rootSignatures);
   }
 
   /**
-   * This method adds implicit edges of the entry points of the call graph algorithm. It will add an
-   * edge to all static initializer of the entry points.
+   * This method returns a list of static initializers that should be considered by the given entry
+   * points
    *
    * @param entryPoints the entry points of the call graph algorithm
-   * @param cg the call graph which will save the added implicit edges.
-   * @param workList the implicit targets will be added to the work list to process in the call
-   *     graph algorithm
    */
-  protected void addImplicitEdgesOfEntryPoints(
-      List<MethodSignature> entryPoints, MutableCallGraph cg, Deque<MethodSignature> workList) {
-    entryPoints.forEach(
-        methodSignature -> {
-          SootMethod clintMethod =
-              view.getMethod(methodSignature.getDeclClassType().getStaticInitializer())
-                  .orElse(null);
-          if (clintMethod == null) {
-            return;
-          }
-          MethodSignature staticInitSig = clintMethod.getSignature();
-          if (!cg.containsMethod(methodSignature)) {
-            cg.addMethod(methodSignature);
-          }
-          if (!cg.containsMethod(staticInitSig)) {
-            cg.addMethod(staticInitSig);
-          }
-          if (!cg.containsCall(methodSignature, staticInitSig)) {
-            cg.addCall(methodSignature, staticInitSig);
-            workList.push(staticInitSig);
-          }
-        });
+  protected List<MethodSignature> getClinitFromEntryPoints(List<MethodSignature> entryPoints) {
+    return entryPoints.stream()
+        .map(
+            methodSignature ->
+                getSignatureOfImplementedStaticInitializer(methodSignature.getDeclClassType()))
```

Comment:
```suggestion
     methodSignature.getDeclClassType().getStaticInitializer())
```
can we do this or do we have to check if the clinit exists? 
If no, maybe refactor/simplify the method name to "getStaticInitializer" ?

Reply:
If the class wont have any static fields which are set, it wont have a clinit method

Reply:
so no we need the check if the method exists, since getSTaticInitializer just returns a signature